caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java

```diff
@@ -0,0 +1,273 @@
+
+    @Override
+    public void setTenant(String tenantId) {
+        //  The tenant identifier is used as part of the Elasticsearch index name.
+        //  There are restrictions around the naming of the index including it must be lowercase
+        //  and not contain commas.
+        if(tenantId.contains(",")) {
+            //  Report invalid comma usage.
+            String errorMessage = "Invalid characters (i.e commas) in the tenant identifier: " + tenantId;
+            LOG.error(errorMessage);
+            throw new IllegalArgumentException(errorMessage);
+        }
+
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.TENANT_ID_FIELD, tenantId.toLowerCase());
+    }
+
+    @Override
+    public void setCorrelationId(String correlationId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.CORRELATION_ID_FIELD, correlationId);
+    }
+
+    @Override
+    public void setEventType(String eventCategoryId, String eventTypeId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_CATEGORY_ID_FIELD, eventCategoryId);
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_TYPE_ID_FIELD, eventTypeId);
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, String value) {
+        auditEventParams.add(new EventParam(name, "string", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, short value) {
+        auditEventParams.add(new EventParam(name, "short", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, int value) {
+        auditEventParams.add(new EventParam(name, "int", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, long value) {
+        auditEventParams.add(new EventParam(name, "long", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, float value) {
+        auditEventParams.add(new EventParam(name, "float", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, double value) {
+        auditEventParams.add(new EventParam(name, "double", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, boolean value) {
+        auditEventParams.add(new EventParam(name, "boolean", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, Date value) {
+        auditEventParams.add(new EventParam(name, "date", columnName, value));
+    }
+
+    @Override
+    public void send() throws Exception {
+
+        //  Get the constructed Audit Event as a JSON string
+        String auditEventJson = getAuditEventAsJsonString();
+
+        // Reset the JsonBuilder
+        jsonBuilder = XContentFactory.jsonBuilder();
```

Comment:
The idea here is that it is being reset so that the user can use the event builder instance again after calling send. Moved this into the method that returns the json string in the latest commit.